File: app/src/main/java/com/mrboomdev/awery/app/data/db/dao/MediaProgressDao.java
package com.mrboomdev.awery.app.data.db.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.mrboomdev.awery.extensions.data.CatalogMediaProgress;

import java.util.List;

@Dao
public interface MediaProgressDao {

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	void insert(CatalogMediaProgress progress);

	@Query("SELECT * FROM media_progress WHERE global_id = :globalId")
	CatalogMediaProgress get(String globalId);

	@Query("SELECT * FROM media_progress WHERE lists LIKE '%;;;' || :list || ';;;%'")
	List<CatalogMediaProgress> getAllFromList(String list);
}

File: .old-shared/src/commonMain/kotlin/com/mrboomdev/awery/data/dao/MediaProgressDao.kt
package com.mrboomdev.awery.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.mrboomdev.awery.data.entity.DBMediaProgress

@Dao
interface MediaProgressDao {
	@Insert(onConflict = OnConflictStrategy.REPLACE)
	suspend fun insert(progress: DBMediaProgress)
	
	@Query("SELECT * FROM DBMediaProgress WHERE globalId = :globalId")
	suspend fun get(globalId: String): DBMediaProgress
	
	@Query("SELECT * FROM DBMediaProgress WHERE lists LIKE '%;;;' || :list || ';;;%'")
	suspend fun getAllFromList(list: String): List<DBMediaProgress>
}